app/app.go

```diff
@@ -446,6 +550,20 @@ func New(
 	app.SetAnteHandler(anteHandler)
 	app.SetEndBlocker(app.EndBlocker)
 
+	// upgradeInfo, err := app.UpgradeKeeper.ReadUpgradeInfoFromDisk()
+	// if err != nil {
+	// 	panic(err)
+	// }
+
+	// if upgradeInfo.Name == "moneta" && !app.UpgradeKeeper.IsSkipHeight(upgradeInfo.Height) {
+	// 	// storeUpgrades := store.StoreUpgrades{
+	// 	// 	Added: []string{authz.ModuleName, feegrant.ModuleName, wasm.ModuleName},
+	// 	// }
+
+	// 	// // configure store loader that checks if version == upgradeHeight and applies store upgrades
+	// 	// app.SetStoreLoader(upgradetypes.UpgradeStoreLoader(upgradeInfo.Height, &storeUpgrades))
```

Comment:
I was unsure if this code was necessary, if it is was just specific to the [Juno moneta](https://github.com/CosmosContracts/juno/issues/87) upgrade, so I left it in but commented it out. If it is unnecessary, we can delete it.

Reply:
we might be able to use it to upgrade, iirc juno did the same, launched a chain and then upgraded to using smart contracts

Reply:
yah.. it's necessary... we need to produce new merkle trees for new modules...